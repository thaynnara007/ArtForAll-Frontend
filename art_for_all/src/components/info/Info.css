.note{
    max-width: 430px;
    border: none;
}

details {
    background: cornflowerblue;
    padding: 1em;
    border: 1px solid hsla(162, 76%, 32%, .3);
    color: black;
    border-radius: 5px;
}

details > p {
    background: white;
    padding: 0.5em 0.7em;
    border-radius: 5px;
    box-shadow: 1px 1px 3px hsla(162, 76%, 32%, .2);
    font-size: 1em;
    margin-top: -0.5em;
    color: dimgrey;
}

.label-style{
    margin-top: 0.2em;
    margin-left: 0.3em;
    color: dimgrey;
}

.summary-margin{
    margin-bottom: 0.4em;
}
.back{
    margin-top: 0.8em;
    background: powderblue;
    border-radius: 5px;
}

.font{
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

