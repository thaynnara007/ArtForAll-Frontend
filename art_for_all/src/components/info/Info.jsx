import React from 'react';
import TagBox from '../tagBox/TagBox';
import './Info.css';
import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';

class Info extends React.Component{

    render(){

        const ART_NAME = 0;
        const AUTHOR = 1;
        const COLLECTION = 2;
        const DESCRIPTION = 3;

        return(
            <div className="container note">
                <details className="font">
                    <summary className="font-weight-bold summary-margin">Informations</summary>
                    <div className="back">
                        <label className="label-style">Art Name:</label>
                    </div>
                    <p>Adventure ends</p>
                    <div className="back ">
                        <label className="label-style">Author:</label>
                    </div>
                    <p>thay@ra</p>
                    <div className="back">
                        <label className="label-style">Collection:</label>
                    </div>
                    <p>Adventure time</p>
                    <div className="back">
                        <label className="label-style">Description:</label>
                    </div>
                    <p>Description: its a art to express the end of the cartoon</p>
                    <div className="back">
                        <label className="label-style">Contact:</label>
                    </div>
                    <p>thaynnara.goncalves@ccc.ufcg.edu.br</p>
                </details>
                <TagBox></TagBox>
            </div>
        )
    }
}

export default Info;