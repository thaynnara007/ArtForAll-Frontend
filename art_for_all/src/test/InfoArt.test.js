import React from 'react';
import Jest from 'jest';
import Enzyme from './Enzyme';
import {shallow} from 'enzyme';
import Info from '../components/info/InfoArt';

describe("INFO ART- testing state screnshot", () =>{

    it("TEST01: it should render correctly", () =>{
        const info = shallow(<Info info={['','','','', '']} tagsName={[]}></Info>);
        expect(info).toMatchSnapshot();
    })

    it("TEST02: it should render correctly with info", () =>{
        const info = shallow(<Info info={['aaaa','bbbb','cccc','ddddd', 'eeee']} tagsName={[]}></Info>);
        expect(info).toMatchSnapshot();
    })
})