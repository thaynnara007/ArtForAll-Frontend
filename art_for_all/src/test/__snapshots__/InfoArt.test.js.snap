// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INFO ART- testing state screnshot TEST01: it should render correctly 1`] = `
<div
  className="container note"
>
  <details
    className="font"
  >
    <summary
      className="font-weight-bold summary-margin"
    >
      Informations
    </summary>
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Art Name:
      </label>
    </div>
    <p />
    <div
      className="back "
    >
      <label
        className="label-style"
      >
        Author:
      </label>
    </div>
    <p />
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Collection:
      </label>
    </div>
    <p />
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Description:
      </label>
    </div>
    <p />
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Contact:
      </label>
    </div>
    <p />
  </details>
  <TagBox
    tagsName={Array []}
  />
</div>
`;

exports[`INFO ART- testing state screnshot TEST02: it should render correctly with info 1`] = `
<div
  className="container note"
>
  <details
    className="font"
  >
    <summary
      className="font-weight-bold summary-margin"
    >
      Informations
    </summary>
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Art Name:
      </label>
    </div>
    <p>
      aaaa
    </p>
    <div
      className="back "
    >
      <label
        className="label-style"
      >
        Author:
      </label>
    </div>
    <p>
      bbbb
    </p>
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Collection:
      </label>
    </div>
    <p>
      cccc
    </p>
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Description:
      </label>
    </div>
    <p>
      ddddd
    </p>
    <div
      className="back"
    >
      <label
        className="label-style"
      >
        Contact:
      </label>
    </div>
    <p>
      eeee
    </p>
  </details>
  <TagBox
    tagsName={Array []}
  />
</div>
`;
